types/decimal_test.go

```diff
@@ -21,6 +22,31 @@ func TestDecimalTestSuite(t *testing.T) {
 	suite.Run(t, new(decimalTestSuite))
 }
 
+func TestDecApproxEq(t *testing.T) {
+
+	// d1 = 5.5, d2 = 6, tol = 0.1
+	d1 := sdk.NewDecWithPrec(55, 2)
+	d2 := sdk.NewDecWithPrec(6, 1)
+	tol := sdk.NewDecWithPrec(1, 1)
+
+	assert.True(sdk.DecApproxEq(t, d1, d2, tol))
+
+	// d1 = 5.5, d2 = 6, tol = 1E-5
+	d1 = sdk.NewDecWithPrec(55, 2)
+	d2 = sdk.NewDecWithPrec(6, 1)
+	tol = sdk.NewDecWithPrec(1, 5)
+
+	assert.False(sdk.DecApproxEq(t, d1, d2, tol))
+
+	// d1 = 6, d2 = 6.1, tol = 0.1
+	d1 = sdk.NewDecWithPrec(6, 1)
+	d2 = sdk.NewDecWithPrec(61, 2)
```

Comment:
aren't these two 0.6 and 0.61?

Reply:
oh yeah sorry, the comments should say 0.6 and 0.61, not 6 and 6.1.

Reply:
Yes, `NewDecWithPrec(a, b)` does return `a * 10^-b`. The constructor is def not the friendliest